src/Dotnet.Script.Core/ScriptDownloader.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Dotnet.Script.Core
+{
+    public class ScriptDownloader
+    {
+        public async Task<string> Download(string uri)
+        {
+            using (HttpClient client = new HttpClient())
+            {
+                using (HttpResponseMessage response = await client.GetAsync(uri))
+                {
+                    response.EnsureSuccessStatusCode();
+                    using (HttpContent content = response.Content)
+                    {
+                        string scriptFile = $"{Path.GetTempFileName()}.csx";                        
+                        var code = await content.ReadAsStringAsync();
+                        File.WriteAllText(scriptFile, code);
```

Comment:
do we need to write the file? we could - I think - just buffer it in memory and call `RunCode()` (the eval path)

Reply:
actually, when we have a file, we could debug. however you'd need to know where the file is, so not sure how useful that is. Maybe in combination with the dotnet-script extension that could work.

the advantage of not having the file is no need to clean up anything. For example at the moment if anything throws, looks like the file would not be cleaned, so we should at least add that if we don't stay in memory